src/types.rs

```diff
@@ -1,6 +1,30 @@
 use crate::Error;
 use derive_builder::Builder;
 
+use std::net::IpAddr;
+use ux::u24;
```

Comment:
u24 is used here only for the ttl in the SdServiceEntry.. it could also become a u32 with only first 3 bytes serialized/parsed (as done in eg the C++ embedded stack)?!

Reply:
done,- use u32 for the ttl.